gensim/models/word2vec.py

```diff
@@ -1090,6 +1090,13 @@ def update_weights(self):
         for i in xrange(len(self.wv.syn0), len(self.wv.vocab)):
             # construct deterministic seed from word AND seed argument
             newsyn0[i-len(self.wv.syn0)] = self.seeded_vector(self.wv.index2word[i] + str(self.seed))
+
+        # Raise an error if an online update is run before initial training on a corpus
+        if not len(self.wv.syn0):
+            raise RuntimeError("You can do an online update of vocabulary on a pre-trained model. " \
```

Comment:
Note it isn't necessary any training has happened - just that no initial/prior vocabulary-discovery has happened (and the `update=True` case assumes that it has). So perhaps instead: "Cannot update vocabulary of model which has no prior vocabulary."